src/main/kotlin/gov/nist/secauto/oscal/tools/server/OscalVerticle.kt

```diff
@@ -451,4 +458,127 @@ class OscalVerticle : CoroutineVerticle() {
             vertx.deployVerticle(OscalVerticle())
         }
     }
-}
\ No newline at end of file
+    private fun handleQueryFileUpload(ctx: RoutingContext) {
+    launch {
+        try {
+            logger.info("Handling Query file upload request")
+            val body = ctx.body().asString()
+            val expression = ctx.queryParam("expression").firstOrNull()
+
+            if (body.isNotEmpty() && expression != null) {
+                // Create a temporary file
+                val tempFile = Files.createTempFile(oscalDir, "upload", ".tmp")
+                val tempFilePath = tempFile.toAbsolutePath()
+                logger.info("Created temporary file: $tempFilePath")
+
+                // Write the body content to the temporary file
+                tempFile.appendText(body)
+                logger.info("Wrote body content to temporary file")
+
+                val args = mutableListOf("query")
+                args.add("-i")
+                args.add(processUrl(tempFilePath.toString()))
+                args.add("-e")
+                args.add(expression)
+                args.add("-m")
+                args.add("https://raw.githubusercontent.com/usnistgov/OSCAL/refs/heads/main/src/metaschema/oscal_complete_metaschema.xml")
```

Comment:
Open to debate: should we not just pin the version we are using with a submodule instead of remote loading this to make it clear what version we are using, not just `develop` or `main`? I know it is easy, but these convenience approaches with dependencies bite me in the rear quite often.

Reply:
We can actually remove this line altogether I'm using the Oscar static context directly and not using this atm

Reply:
OK feel free to resolve apparently I cannot. lol

Reply:
removed